yt/yt/ytlib/api/native/client_job_info_impl.cpp

```diff
@@ -1458,6 +1477,10 @@ static std::vector<TJob> ParseJobsFromArchiveResponse(
             job.Address = record.Address.value_or("");
         }
 
+        if (responseIdMapping.Addresses) {
+            job.Addresses = ConvertTo<TAddressMap>(record.Addresses.value_or(TYsonString(TString("{}"))));
```

Comment:
To my taste, an explicit `if (record.Addresses)` would be better.

We check `responseIdMapping` to know if the field has been requested in the first place. This is needed for fields that are mandatory, so we must put some default value. In this case, I am not sure we want to make this new field mandatory.